staging/src/k8s.io/client-go/tools/clientcmd/loader.go

```diff
@@ -40,13 +40,15 @@ import (
 const (
 	RecommendedConfigPathFlag   = "kubeconfig"
 	RecommendedConfigPathEnvVar = "KUBECONFIG"
-	RecommendedHomeDir          = ".kube"
+	RecommendedHomeDir          = ".config"
+	RecommendedKubeDir          = "kube"
 	RecommendedFileName         = "config"
 	RecommendedSchemaName       = "schema"
+	OldRecommendedHomeDir       = ".kube"
 )
 
 var (
-	RecommendedConfigDir  = filepath.Join(homedir.HomeDir(), RecommendedHomeDir)
+	RecommendedConfigDir  = filepath.Join(homedir.HomeDir(), RecommendedHomeDir, RecommendedKubeDir)
```

Comment:
changing this value will break many client-go consumers... I don't see a discussion of how this will be avoided/mitigated (the migration below is not sufficient ... see the comment on that function)

Reply:
@liggitt  Yes, I was aware that it might have a documented change to all users. Probably the simplest way is replace the default value: ~/.kube/.kubeconfig to ~/.config/kube/config. Please let me know your thoughts.

Reply:
this discussion should be happening in the design proposal, calling out interactions between the matrix of the following inputs:

environment:
* overridden $KUBECONFIG
* no $KUBECONFIG envvar

clients:
* old client-go consumer
* old client-go writer
* new client-go consumer
* new client-go writer
* non-client-go consumer
* non-client-go writer

Reply:
@liggitt Thanks again for your input. Updated the patch, now going to update the KEP.